bitcoind fails to build on macOS

**Describe the bug**

nix-shell -p bitcoind fails with:

```
…
checking for exit in -lboost_filesystem... yes
checking whether the Boost::Thread library is available... yes
checking for exit in -lboost_thread... yes
checking whether the Boost::Chrono library is available... yes
checking for exit in -lboost_chrono... yes
configure: error: hexdump is required for tests
builder for '/nix/store/dsxli899j1239pr51sb5728k2r6hj0l3-bitcoind-0.19.0.1.drv' failed with exit code 1
error: build of '/nix/store/dsxli899j1239pr51sb5728k2r6hj0l3-bitcoind-0.19.0.1.drv' failed
```

**Metadata**
Please run `nix run nixpkgs.nix-info -c nix-info -m` and paste the result.

```
$ nix run nixpkgs.nix-info -c nix-info -m
error: attribute 'nixpkgs' in selection path 'nixpkgs.nix-info' not found
```

(I'm running nixpkgs-unstable, and I just updated the channel)

Maintainer information:
```yaml
attribute: bitcoind
```

--- a/pkgs/applications/blockchains/bitcoin.nix
+++ b/pkgs/applications/blockchains/bitcoin.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchurl, pkgconfig, autoreconfHook, openssl, db48, boost, zeromq, rapidcheck
+{ stdenv, fetchurl, pkgconfig, autoreconfHook, openssl, db48, boost, zeromq, rapidcheck, hexdump
 , zlib, miniupnpc, qtbase ? null, qttools ? null, wrapQtAppsHook ? null, utillinux, python3, qrencode, libevent
 , withGui }:
 
@@ -31,6 +31,7 @@ in stdenv.mkDerivation rec {
 
   nativeBuildInputs =
     [ pkgconfig autoreconfHook ]
+    ++ optional stdenv.isDarwin hexdump
     ++ optional withGui wrapQtAppsHook;
   buildInputs = [ openssl db48 boost zlib zeromq
                   miniupnpc libevent]
@@ -75,7 +76,6 @@ in stdenv.mkDerivation rec {
     homepage = http://www.bitcoin.org/;
     maintainers = with maintainers; [ roconnor AndersonTorres ];
     license = licenses.mit;
-    # bitcoin needs hexdump to build, which doesn't seem to build on darwin at the moment.
-    platforms = platforms.linux;
+    platforms = platforms.unix;
   };
 }

